pkg/k8s/rule_translate.go

```diff
@@ -186,8 +195,10 @@ func deleteToCidrFromEndpoint(
 				return err
 			}
 			// delete all generated CIDRs for a CIDR that match the given
-			// endpoint
-			if c.Generated && cidr.Contains(epIP) {
+			// endpoint.
+			// Only delete CIDR that has exactly the same IP as 'epIP', to avoid
+			// deleting CIDRs sharing a prefix with 'epIP'.
+			if cidr.Contains(epIP) && cidr.IP.Equal(epIP) {
```

Comment:
`cidr.IP` is `10.0.0.0` so it will never be equal to the `epIP`

Reply:
This function is deleting full/one-address CIDRs only, so the `cidr.IP` will be `10.0.0.1` for a backend with IP `10.0.0.1`.

Reply:
Given that this is only used from the ToServices -> ToCIDR generation code, I am inclined to agree with you, though I find it misleading that this code is using `cidr.Contains()` and then making assumptions that the CIDRs are fully-specified. We should probably raise some loud warnings if this constraint is ever violated.

Reply:
@jrajahalme you are right in here as well. I think my confusion was that I tested things out with a `/24` cidr and not a `/32`.